File: day10/ex3.py
class Student:
    def __init__(self, roll_no, name, marks):
        self.roll_no = roll_no
        self.name = name
        self.marks = marks

    def calculate_total(self):
        return sum(self.marks)

    def calculate_average(self):
        return self.calculate_total() / len(self.marks)

    def calculate_grade(self):
        avg = self.calculate_average()

        if avg >= 90:
            return "A"
        elif avg >= 75:
            return "B"
        elif avg >= 60:
            return "C"
        elif avg >= 40:
            return "D"
        else:
            return "F"

    def display_report(self):
        print("\n------ Report Card ------")
        print("Roll Number:", self.roll_no)
        print("Student Name:", self.name)

        for i in range(5):
            print(f"Subject {i+1}: {self.marks[i]}")

        print("Total Marks:", self.calculate_total())
        print("Average:", self.calculate_average())
        print("Grade:", self.calculate_grade())


roll = input("Enter Roll Number: ")
name = input("Enter Student Name: ")

marks = []
for i in range(5):
    mark = float(input(f"Enter Marks of Subject {i+1}: "))
    marks.append(mark)

student = Student(roll, name, marks)
student.display_report()

File: day4/ex3.js
const students = [
    {
        name: "Alice",
        marks: [80, 85, 90]
    },
    {
        name: "Bob",
        marks: [60, 70, 75]
    },
    {
        name: "Charlie",
        marks: [95, 92, 98]
    }
];
let topper = "";
let highestAvg = 0;
let above80 = [];
students.forEach(student => {
    let total = 0;
    student.marks.forEach(mark => total += mark);
    let avg = total / student.marks.length;
    console.log(student.name, "Average:", avg);
    if (avg > highestAvg) {
        highestAvg = avg;
        topper = student.name;
    }
    if (avg > 80) {
        above80.push(student.name);
    }
});
console.log("Topper:", topper);
console.log("Above 80:", above80);